### DISCUSSION TITLE
Feature Request: Ability to Copy citation to Clipboard

### DISCUSSION THREAD

**Discussion by Pottal** [quarto-dev/quarto-cli#8788] [Feature Requests]
## Description
I am using Quarto for creating websites and blogs. To facilitate easy sharing of article titles and URLs, I have customized the front matter with appendix-cite-as: display and also customized the CSL to display the article's title and URL at the end of the articles. However, this requires users to manually copy and paste the information if they wish to share it. I would like to propose a feature to make sharing even simpler.

## Problem Statement
While the current setup with appendix-cite-as: display and customized CSL works for displaying the article's title and URL at the end of an article, it doesn't offer a seamless way for users to share this information. Users have to manually select the text and copy it, which can be cumbersome and not user-friendly.

## Proposed Solution
1. **Button to Copy to Clipboard**: Implement a button functionality that allows users to easily copy the article's title and URL to their clipboard with a single click. This button could be placed near the displayed title and URL at the end of the article.

2. **Code Block for Citation**: Alternatively, or additionally, format after the "For attribution, please cite this work as:" section and its contents into a code block. This would make it clearer for users that this section is meant to be copied and would facilitate easier copying and pasting. (The code block comes with a button to copy it to the clipboard, making it easy to copy.)

## Additional Context
This feature would significantly enhance user experience by simplifying the process of sharing articles created with Quarto. It would also align with the goals of making scholarly and professional communication more accessible and user-friendly.

![image](https://github.com/quarto-dev/quarto-cli/assets/108885499/95b76884-d41f-4b36-a771-0a50ad53b5c9)

**Comment by mcanouil** [comment]
Could you provide a concrete example of where is the issue?

Is your request only about the second block in the below screenshot?

Using `citation: true` does come with a button to copy/paste the citation.
<img width="1624" alt="image" src="https://github.com/quarto-dev/quarto-cli/assets/8896044/bc74f0ee-9e8b-4e37-a506-244ba00e28ac">

**Reply by Pottal** [reply]
Thank you for your reply. As you mentioned, my request pertains only to the second block. Since it is for a blog and not an academic paper, I do not intend to use BibTeX citations. Instead, I requested this because I would appreciate a feature that is more shareable.

**Reply by mcanouil** [reply]
Note that the usual way to integrate citation even in Word document is through Bibtex or similar tools, because the formatted string is formatted according to a style which cannot easily be changed in a formatted string since the information are not necessarily shown the same way or shown at all.

Anyhow, closing in favour of:
- #8800

**Reply by xuefeng-xu** [reply]
I'm using version 1.8.25, I noticed the copy button in Bibtex now is missing.

<img width="1423" height="521" alt="image" src="https://github.com/user-attachments/assets/48d680bc-a40d-40cb-b88e-139f3a609072" />

**Reply by mcanouil** [reply]
@xuefeng-xu this discussion is closed and is about something else. Please open a new issue with a fully reproducible example to replicate the issue you want to report.

> [!important]
> Accessibility: To improve accessibility, please add alternative text to your screenshots. This helps all users, including those using screen readers, to understand the context of the images. A brief description can make a big difference! See [Good Alt Text, Bad Alt Text — Making Your Content Perceivable](https://www.wcag.com/blog/good-alt-text-bad-alt-text-making-your-content-perceivable/).
